### PR TITLE
added Phi-3 handlers

### PR DESCRIPTION
**Pull request by AndyChenYH**
Added handler for:
phi-3-mini-4k-instruct
phi-3-mini-128k-instruct
phi-3-small-8k-instruct
phi-3-small-128k-instruct
phi-3-medium-4,-instruct
phi-3-medium-128k-instruct
phi-3.5-mini-instruct

|Rank|Model                            |Model Link                                               |Organization|License     |AST Summary|Simple AST|Multiple AST|Parallel AST|Parallel Multiple AST|Irrelevance Detection|Relevance Detection|
|----|---------------------------------|---------------------------------------------------------|------------|------------|-----------|----------|------------|------------|---------------------|---------------------|-------------------|
|1   |Phi-3-small-8k-instruct (Prompt) |https://huggingface.co/microsoft/Phi-3-small-8k-instruct |Microsoft   |MIT         |66.39%     |59.70%    |64.20%      |76.75%      |64.92%               |47.06%               |87.80%             |
|2   |Phi-3-medium-4k-instruct (Prompt)|https://huggingface.co/microsoft/Phi-3-medium-4k-instruct|Microsoft   |MIT         |62.10%     |66.67%    |67.40%      |62.00%      |52.33%               |46.79%               |78.05%             |
|3   |Phi-3-mini-4k-instruct (Prompt)  |https://huggingface.co/microsoft/Phi-3-mini-4k-instruct  |Microsoft   |MIT         |66.63%     |70.76%    |75.67%      |69.75%      |50.33%               |20.25%               |75.61%             |
|4   |Phi-3.5-mini-instruct (Prompt)   |https://huggingface.co/microsoft/Phi-3.5-mini-instruct   |Microsoft   |MIT         |55.13%     |64.22%    |66.12%      |52.00%      |38.17%               |64.93%               |70.73%             |
|5   |Phi-3-mini-128k-instruct (Prompt)|https://huggingface.co/microsoft/Phi-3-mini-128k-instruct|Microsoft   |MIT         |51.49%     |67.60%    |72.50%      |41.12%      |24.75%               |44.07%               |85.37%             |

### CODE REVIEW

File: berkeley-function-call-leaderboard/bfcl/model_handler/oss_model/phi.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class PhiHandler(OSSHandler):
+    def __init__(self, model_name, temperature=0.001, top_p=1, max_tokens=1000) -> None:
+        super().__init__(model_name, temperature, top_p, max_tokens)
+
+    def _format_prompt(self, prompts, function, test_category):
+        formatted_prompt = ""
+        for prompt in prompts:
+            role_token = f"<|{prompt['role']}|>" if prompt['role'] in ['system', 'user', 'assistant'] else "<|system|>"
```
Comment (HuanzhiMao) [inline berkeley-function-call-leaderboard/bfcl/model_handler/oss_model/phi.py:12]: You missed the `\n` at the end. 

> {% for message in messages %}{% if message['role'] == 'system' and message['content'] %}{{'<\|system\|>\n' + message['content'] + '<\|end\|>\n'}}{% elif message['role'] == 'user' %}{{'<\|user\|>\n' + message['content'] + '<\|end\|>\n'}}{% elif message['role'] == 'assistant' %}{{'<\|assistant\|>\n' + message['content'] + '<\|end\|>\n'}}{% endif %}{% endfor %}{% if add_generation_prompt %}{{ '<\|assistant\|>\n' }}{% else %}{{ eos_token }}{% endif %}

**Comment by HuanzhiMao** [comment]
Also, `Phi-3-mini`, `Phi-3-small`, `Phi-3-medium`, and `Phi-3.5` all have different chat templates. So you need different `_format_prompt` functions.

**Review by HuanzhiMao — APPROVED** [review]
LGTM

**Review by ShishirPatil — APPROVED** [review]
Thank you @AndyChenYH and congratulations on landing your first PR!

**Pull request merged**